crates/sui/src/upgrade_compatibility.rs

```diff
@@ -496,30 +497,19 @@ fn missing_definition_diag(
     declaration_kind: &str,
     identifier_name: &Identifier,
     compiled_unit_with_source: &CompiledUnitWithSource,
-    file_id: usize,
-) -> Diagnostic<usize> {
+) -> Diagnostic {
     let module_name = compiled_unit_with_source.unit.name;
-
-    let start = compiled_unit_with_source
-        .unit
-        .source_map
-        .definition_location
-        .start() as usize;
-
-    let end = compiled_unit_with_source
+    let loc = compiled_unit_with_source
         .unit
         .source_map
-        .definition_location
-        .end() as usize;
-
-    Diagnostic::error()
-        .with_message(format!("{declaration_kind} is missing"))
-        .with_labels(vec![Label::primary(file_id, start..end).with_message(
-            format!(
-                "Module '{module_name}' expected {declaration_kind} '{identifier_name}', but found none"
-            ),
-        )])
-        .with_notes(vec![format!(
-            "{declaration_kind}s are part of a module's public interface and cannot be removed or changed during an upgrade, add back the {declaration_kind} '{identifier_name}'."
-        )])
+        .definition_location;
+
+    Diagnostic::new(
```

Comment:
Is the `diag` macro exported from the compiler? You might find that easier to use (except for notes... maybe we should fix that)

Reply:
I didn't come across this but I'll look around.

Reply:
maybe a positional `notes = xyz`